src/ixdat/db.py

```diff
@@ -0,0 +1,229 @@
+        self._id = None  # SHOULD BE SET AFTER THE __INIT__ OF INHERITING CLASSES
+        self.name = None  # MUST BE SET IN THE __INIT__ OF INHERITING CLASSES
+
+    def __repr__(self):
+        return f"{self.__class__}(id={self.id}, name={self.name})"
+
+    @property
+    def id(self):
+        """The principle-key identifier. Set by backend or counted in memory."""
+        if not self._id:
+            if self.backend_name == "memory":
+                self._id = memory_backend.get_next_available_id(self.table_name)
+            else:
+                raise DataBaseError(
+                    f"{self} comes from {self.backend_name} "
+                    f"but did not get an id from its backend."
+                )
+        return self._id
+
+    @id.setter
+    def id(self, i):
+        """Backends should explicitly set obj.id after loading a Saveable obj"""
+        self._id = i
+
+    def get_main_dict(self):
+        """Return dict: serializition only of the row of the object's main table"""
+        if self.column_attrs is None:
+            raise DataBaseError(
+                f"{self} can't be seriealized because the class {self.__class__} "
+                f"hasn't defined column_attrs"
+            )
+        self_as_dict = {
+            column: getattr(self, attr) for column, attr in self.column_attrs.items()
+        }
+        return self_as_dict
+
+    def as_dict(self):
+        """Return dict: serialization of the object main and auxiliary tables"""
+        main = self.get_main_dict()
+        self_as_dict = main
+        if self.extra_column_attrs:
+            aux_tables_dict = {
+                table_name: {column: getattr(self, attr) for column, attr in extras}
+                for table_name, extras in self.extra_column_attrs.items()
+            }
+            for aux_dict in aux_tables_dict.values():
+                self_as_dict.update(**aux_dict)
+        if self.extra_linkers:
+            linker_tables_dict = {
+                table_name: {column: getattr(self, attr) for column, attr in linkers}
+                for table_name, linkers in self.extra_linkers.items()
+            }
+            for linker in linker_tables_dict.values():
+                self_as_dict.update(**linker[1])
+        return self_as_dict
+
+    def save(self):
+        """Save self and return the id. This sets self.backend_name and self.id"""
+        return self.db.save(self)
+
+    @classmethod
+    def from_dict(cls, obj_as_dict):
+        """Return an object built from its serialization."""
+        return cls(**obj_as_dict)
+
+    @classmethod
+    def open(cls, i):
+        """Open an object given its id (the table is cls.table_name)"""
+        return cls.db.open(cls, i)
+
+    def load_data(self):
+        """Load the data of the object, if ixdat in its laziness hasn't done so yet"""
+        return self.db.load_obj_data(self)
+
+
+class PlaceHolderObject:
```

Comment:
Yes, this is used by `Measurement`. 
Maybe it should be used in `ValueSeries` and `Field` as well. That could avoid any need of `_t_id` and `_axis_ids`...

Reply:
Put in a TODO.

Reply:
Ok to resolve for me, but I wont do it if you use the unresolved for some kind of todo.

Reply:
Okay, will resolve since the info is now in the code. I've been using unresolved more as indicating what I still want you to look at than todo. But not sure what the convention on github is with that.